examples/cms-starter/src/App.tsx

```diff
@@ -0,0 +1,160 @@
+import { framer } from "framer-plugin"
+import "./App.css"
+import { useLayoutEffect, useState } from "react"
+import {
+    COLLECTIONS_SYNC_MAP,
+    computeFieldConfig,
+    DataSource,
+    FieldConfig,
+    getDataSources,
+    listDataSourcesIds,
+    LOCAL_STORAGE_LAST_LAUNCH_KEY,
+    PLUGIN_COLLECTION_SYNC_REFERENCE_KEY,
+    PLUGIN_COLLECTION_SYNC_SLUG_KEY,
+    syncCollection,
+} from "./data"
+import { FieldMapping } from "./FieldMapping"
+
+const activeCollection = await framer.getManagedCollection()
+const existingFields = activeCollection ? await activeCollection.getFields() : []
+
+const syncDataSourceId = await activeCollection.getPluginData(PLUGIN_COLLECTION_SYNC_REFERENCE_KEY)
+const syncSlugFieldId = await activeCollection.getPluginData(PLUGIN_COLLECTION_SYNC_SLUG_KEY)
+
+const syncDataSource = syncDataSourceId ? await getDataSources(syncDataSourceId) : null
+
+let savedFieldsConfig: FieldConfig[] | undefined
+
+if (syncDataSource) {
+    savedFieldsConfig = computeFieldConfig(existingFields, syncDataSource)
+}
+
+if (framer.mode === "syncManagedCollection" && savedFieldsConfig && syncDataSource && syncSlugFieldId) {
+    await syncCollection(
+        syncDataSource,
+        savedFieldsConfig
+            .filter(field => field.field && !field.source.ignored)
+            .filter(field => !field.reference || field.reference.destination !== null),
+        syncSlugFieldId
+    )
```

Comment:
Should this have a return here?

Reply:
This is not a function so we can't really return, but `syncCollection` will call `framer.closePlugin` on success